mlir-sycl/test/Conversion/SYCLToLLVM/sycl-constructors-to-llvm.mlir

```diff
@@ -0,0 +1,254 @@
+// RUN: sycl-mlir-opt -convert-sycl-to-llvm='use-opaque-pointers=1' -reconcile-unrealized-casts %s | FileCheck %s
```

Comment:
`-reconcile-unrealized-cast` removes the trivial `unrealized_conversion_cast` operations we introduce